engine/src/main/java/io/camunda/zeebe/engine/processing/deployment/model/validation/ZeebeTaskHeadersValidator.java

```diff
@@ -40,4 +46,18 @@ private void checkForReservedHeaderKeys(
         .map(key -> String.format(RESERVED_KEY_MESSAGE_TEMPLATE, key, RESERVED_HEADER_NAME_PREFIX))
         .forEach(message -> validationResultCollector.addError(0, message));
   }
+
+  private void checkForDuplicateKeys(
+      final ZeebeTaskHeaders element, final ValidationResultCollector validationResultCollector) {
+    element.getHeaders().stream()
+        .map(ZeebeHeader::getKey)
+        .collect(Collectors.toMap(Function.identity(), value -> 1, Integer::sum))
```

Comment:
🔧 Not sure if this is better, but alternatively you could use:
```.collect(Collectors.groupingBy(Function.identity(), Collectors.counting()))```

Reply:
Turns out this one behaves differently with respect to null values. I will revert back to the first way.
https://ci.zeebe.camunda.cloud/job/camunda/job/zeebe/job/9416-duplicate-task-headers/2/testReport/junit/io.camunda.zeebe.engine.processing.bpmn.activity/ActivityTest(java-testrun)/Verify___Test___shouldIgnoreTaskHeadersIfEmpty/